corehq/apps/app_manager/xpath_validator/nodejs/xpathConfig.js

```diff
@@ -0,0 +1,48 @@
+if (typeof require !== 'undefined' && typeof exports !== 'undefined') {
+    var parser = require('js-xpath/parser');
+    var xpath = require('js-xpath/xpath');
+}
+
+var XPATH_CONFIG = (function () {
+    function getAllowedHashtags(allowCaseHashtags) {
+        var replacements = {
+        '#session': "ok",
+        '#user': "ok"
+        };
+        if (allowCaseHashtags) {
+            replacements['#case'] = 'ok';
+            replacements['#parent'] = 'ok';
+            replacements['#host'] = 'ok';
+        }
+        return replacements;
+    }
+
+    function configureHashtags(allowCaseHashtags) {
+        var p = new parser.Parser();
+        p.setXPathModels = function(models) {
+            p.yy.xpathmodels = models;
+        };
+        var replacements = getAllowedHashtags(allowCaseHashtags);
+        p.setXPathModels(xpath.makeXPathModels({
+            isValidNamespace: function (namespace) {
+                return replacements.hasOwnProperty('#' + namespace);
+            },
+            hashtagToXPath: function (hashtagExpr) {
+                if (replacements.hasOwnProperty(hashtagExpr)) {
+                    return replacements[hashtagExpr];
+                } else {
+                    throw new Error("This should be overridden");
+                }
+            },
+            toHashtag: function (xpath_) {
+                return xpath_.toXPath();
```

Comment:
This could throw an error, since it doesn't seem like it would be expected to be used